### DISCUSSION TITLE
Desktop audio capturing

### DISCUSSION THREAD

**Discussion by Fractal-Tess** [obsproject/obs-studio#7930] [General]
Hey there,

I'm very interested in the steps obs takes to capture desktop audio on Linux systems.
I've tried to explore this topic on my own since yesterday because I wanted to build myself a recording software using rust, however, I was astonished to see how difficult it is to capture desktop audio. 

So far from what I understand, on Linux, you are not able to capture desktop audio directly, and you need to set up an input stream and somehow route the desktop audio into it. I'm not sure, which is why I'm here.

Can you please give me a short rundown of how I may be able to capture the desktop audio?

**Comment by jetspiking** [comment]
@Fractal-Tess I came across this guide (specifically for Linux):

https://0pointer.de/blog/projects/guide-to-sound-apis

This seems to contain a lot of information about when to use which API. I don't know the details of your application. However, this could be useful.

**Comment by Fenrirthviti** [comment]
This sounds more likely to be a support request, and we are not currently accepting support requests on GitHub Discussions. Make sure you read any [pinned posts](https://github.com/obsproject/obs-studio/discussions/5206) on discussion boards before opening a new discussion. Please use our [forums](https://obsproject.com/forum/) or [Discord](https://obsproject.com/discord/) for further assistance.

Thank you!
